crates/uv/src/commands/project/run.rs

```diff
@@ -1075,6 +1078,39 @@ hint: If you are running a script with `{}` in the shebang, you may need to incl
     // TODO(zanieb): Throw a nicer error message if the command is not found
     let handle = process
         .spawn()
+        .map_err(|err| {
+            let executable: Cow<'_, str> = command.display_executable();
+            // Special case for providing meaningful error message when users 
+            // attempt to invoke python. E.g. "python3.11". 
+            // Will not work if patch version is provided. I.E. "python3.11.9"
+            if err.kind() == std::io::ErrorKind::NotFound && is_python_executable(&executable) {
+                // Get version from python command string
+                // e.g. python3.12 -> "3.12" or "" if python version not specified. 
+                let version_part = executable.strip_prefix("python").unwrap_or("");
```

Comment:
Should `is_python_executable` be `python_executable_version -> Some(str)` instead? Then you can avoid the `unwrap_or`?

Reply:
Applied: https://github.com/astral-sh/uv/pull/12201/commits/0cb2d5e35dddaabf6eef93410643bbc58e8098fd